### PR TITLE
Add documentation paragraph to `rigidbody` module description

### PR DESCRIPTION
**Pull request by joaomcteixeira**
- [x] Your PR is about writing documentation for already existing code :fire:

---

I took the brief explanation of the `it0` step in the `local` tutorial for HADDOCK2.x and modified it slightly to fit as a short explanation for the `rigidbody` module.

Also, updated title in doc pages and added exceptions to doc linting in the pertinent files.

If you generate the docs `tox -e docs`, you'lll see this text renders in the HTML of the rigidbody page.

Feel welcome to edit further.

Cheers,

### CODE REVIEW

**Review by amjjbonvin — APPROVED** [review]
Made some edits in the text - check before merging

**Review by amjjbonvin — APPROVED** [review]

**Comment by amjjbonvin** [comment]
Some linting to correct (was editing directly on the GitHub site...

File: src/haddock/modules/sampling/rigidbody/__init__.py [hunk 1]
```diff
@@ -1,4 +1,31 @@
...
+This module does a **randomization of orientations and rigid-body
+minimization.** It corresponds to the classical ``it0`` step in the HADDOCK2.x
+series.
+
+In this module, the interacting partners are treated as rigid bodies, meaning
+that all geometrical parameters such as bond lengths, bond angles, and dihedral
+angles are frozen. The partners are first separated in space and randomly rotated
+around their respective centres of mass. Afterwards, the molecules are brought 
+together by rigid-body energy minimisation with as only degrees of freedome rotations
+and translation.
```
Comment (mgiulini) [inline src/haddock/modules/sampling/rigidbody/__init__.py:14]:
```suggestion
together by rigid-body energy minimisation with as only rotations and translation
as the only degrees of freedom
```

File: src/haddock/modules/sampling/rigidbody/__init__.py [hunk 2]
```diff
@@ -1,4 +1,31 @@
...
+series.
+
+In this module, the interacting partners are treated as rigid bodies, meaning
+that all geometrical parameters such as bond lengths, bond angles, and dihedral
+angles are frozen. The partners are first separated in space and randomly rotated
+around their respective centres of mass. Afterwards, the molecules are brought 
+together by rigid-body energy minimisation with as only degrees of freedome rotations
+and translation. 
+
+The driving force for this energy minimisation is the energy function which
```
Comment (mgiulini) [inline src/haddock/modules/sampling/rigidbody/__init__.py:16]:
```suggestion
The driving force for this energy minimisation is the energy function, which 
```

File: src/haddock/modules/sampling/rigidbody/__init__.py [hunk 3]
```diff
@@ -1,4 +1,31 @@
...
+angles are frozen. The partners are first separated in space and randomly rotated
+around their respective centres of mass. Afterwards, the molecules are brought 
+together by rigid-body energy minimisation with as only degrees of freedome rotations
+and translation. 
+
+The driving force for this energy minimisation is the energy function which 
+consists of the intermolecular van der Waals and electrostatic energy terms and the
+restraints defined to guide the docking. The restraints are distance-based and can consist
+of unambiguous or ambiguous interactions restraints (AIRS). In ab-initio docking mode those 
+restraintscan be automatically defined in various ways (e.g. between center of masses
```
Comment (mgiulini) [inline src/haddock/modules/sampling/rigidbody/__init__.py:20]:
```suggestion
restraints can be automatically defined in various ways (e.g. between center of masses 
```

File: src/haddock/modules/sampling/rigidbody/__init__.py [hunk 4]
```diff
@@ -1,4 +1,31 @@
...
+restraints defined to guide the docking. The restraints are distance-based and can consist
+of unambiguous or ambiguous interactions restraints (AIRS). In ab-initio docking mode those 
+restraintscan be automatically defined in various ways (e.g. between center of masses 
+(CM restraints) or between randomly selected patches on the surface (random AIRs).
+
+The definition of those restraints is particularly important as they effectively guide 
+the minimisation process. For example, with a stringent set of AIRs or unambiguous distance restraints, 
+the solutions of the minimisation will converge much better and the sampling can be limited. 
+In ab-initio mode, however, very diverse solutions will be obtained and the sampling should be 
+increased to make sure to sample enough the possible interaction space
```
Comment (mgiulini) [inline src/haddock/modules/sampling/rigidbody/__init__.py:27]:
```suggestion
increased to make sure to sample enough the possible interaction space.
```

**Review by mgiulini — APPROVED** [review]
just corrected some minor stuff

**Review by amjjbonvin — APPROVED** [review]

**Comment by joaomcteixeira** [comment]
solved lint and some additional minors. Thanks for reviewing. merging when tests are solved

**Pull request merged**